src/app/core/filter/list-filter/list-filter.component.ts

```diff
@@ -15,28 +15,37 @@ import { MatButtonModule } from "@angular/material/button";
     MatFormFieldModule,
     MatSelectModule,
     ReactiveFormsModule,
-    BorderHighlightDirective,
     MatButtonModule,
     ReactiveFormsModule,
+    BasicAutocompleteComponent,
   ],
 })
-export class ListFilterComponent<E extends Entity> {
+export class ListFilterComponent<E extends Entity> implements OnInit {
   @Input({ transform: (value: any) => value as SelectableFilter<E> })
   filterConfig: SelectableFilter<E>;
 
+  autocompleteControl = new FormControl([]);
+
+  ngOnInit() {
+    this.autocompleteControl.setValue(this.filterConfig.selectedOptionValues);
+
+    this.autocompleteControl.valueChanges.subscribe((values) => {
+      this.filterConfig.selectedOptionChange.emit(values);
+    });
+    this.filterConfig.selectedOptionChange.subscribe((values) => {
+      this.autocompleteControl.setValue(values, { emitEvent: false });
+    });
+  }
+
+  getOptionLabel = (option: any) => option.label;
+  getOptionValue = (option: any) => option.key;
+
   selectAll() {
-    this.filterConfig.selectedOptionValues = this.filterConfig.options.map(
-      (option) => option.key,
-    );
-    this.filterConfig.selectedOptionChange.emit(
-      this.filterConfig.selectedOptionValues,
-    );
+    const allValues = this.filterConfig.options.map((opt) => opt.key);
+    this.autocompleteControl.setValue(allValues);
   }
 
   deselectAll() {
```

Comment:
if I do a "Clear" for an individual filter here, the "Clear all" button still remains visible in the overall list although everything is cleared. (Maybe not directly related to this PR but still a tiny glitch we should fix)

Reply:
Ah, thanks for noticing this , i have fixed this now